src/adapter/src/rbac.rs

```diff
@@ -347,6 +387,42 @@ impl fmt::Display for Attribute {
     }
 }
 
+fn attribute_err(unheld_attributes: Vec<Attribute>, plan: &Plan) -> Result<(), UnauthorizedError> {
+    if !unheld_attributes.is_empty() {
```

Comment:
Super nit, more like a coding preference, so ignore if you like:

```
if unheld_attributes.is_empty() {
    return Ok(());
}
...
```

This pattern reduces overall indentation and IMO increases readability.
```